pkg/apis/common/common.go

```diff
@@ -153,3 +167,13 @@ type Metadata struct {
 	Annotations map[string]string `json:"annotations,omitempty" protobuf:"bytes,1,rep,name=annotations"`
 	Labels      map[string]string `json:"labels,omitempty" protobuf:"bytes,2,rep,name=labels"`
 }
+
+func (s SASLConfig) GetMechanism() sarama.SASLMechanism {
```

Comment:
Overall everything looks good to me except this part.

1. Returning `sarama.SASLMechanism` makes `github.com/Shopify/sarama` get introduced to the models, which is not good. Ideally the dependencies in models are either from native GoLang or k8s related. I prefer to return a string here.
2. Another reason is, since this struct is put in `common`, it is intended to be used by not only Kafka. Assume it's going to be used by something else, `sarma.SASLMechanism` might not be recognized by it. Also, I made a little research on SASL auth, it looks like the mechanism is not unified, different product has different support (PLAIN might be the one everyone supports). Correct me if I'm wrong.

Reply:
1. Apologies for the oversight. Removed the type reference and refactored again to reflect on the actual sensor and eventsource. 
2. I'm basing the authentication mechanism on the available library which is Sarama. The other authentication mechanisms might be something to look into in the future, but "PLAIN" auth is a good place to start.